lib/membrane_vpx/decoder/vpx_decoder.ex

```diff
@@ -103,4 +74,40 @@ defmodule Membrane.VPx.Decoder do
     buffers = Enum.map(decoded_frames, &%Buffer{payload: &1, pts: pts})
     {stream_format_action ++ [buffer: {:output, buffers}], state}
   end
+
+  @spec get_output_stream_format(
+          RemoteStream.t() | VP8.t() | VP9.t(),
+          RawVideo.pixel_format(),
+          State.t()
+        ) :: RawVideo.t()
+  defp get_output_stream_format(
+         input_stream_format,
+         pixel_format,
+         %State{codec_module: codec_module} = state
+       ) do
+    {width, height, framerate} =
+      case input_stream_format do
+        %RemoteStream{} ->
+          {
+            state.width || raise("Width not provided"),
+            state.height || raise("Height not provided"),
+            state.framerate || raise("Framerate not provided")
+          }
+
+        %^codec_module{width: width, height: height, framerate: framerate} ->
```

Comment:
do we really need that `codec_module` here? It's only stored in the state to be used here.
You could match on `%{width: width, height: height, framerate: framerate}` and the stream format definition will make sure that no other module than `VP8` or `VP9` can be applied here.

Reply:
I did this to ensure the stream format is actually matching the codec, but i guess accepted_format is enough